File: CPP_FFFE/21_ReferenceAndValueSemantics/code/02_classes.cpp
#include <iostream>

struct Fraction {
  int numerator, denominator;
  auto operator<=>(Fraction const &) const = default;
};

int main() {
  Fraction a {3, 5};
  Fraction b {a};
  a.numerator = 0;
  std::cout << (a != b ? "value" : "reference")
    << " semantics for classes\n";
}

File: CPP_FFFE/21_ReferenceAndValueSemantics/code/02_classes.py
class Fraction:
    def __init__(self, n, d):
        self.numerator = n
        self.denominator = d

a = Fraction(3, 5)
b = a
a.numerator = 0

method = "value" if a != b else "reference"
print(f"{method} semantics for classes")
